dist/Makefile.tmpl

```diff
@@ -23,18 +23,22 @@
 # By default, this Makefile relies on the local checkout of kremlib
 KREMLIN_HOME ?= ../kremlin
 
+-include Makefile.config
+
 # 1. The usual pseudo auto-configuration
 
-UNAME		= $(shell uname)
-MARCH		= $(shell uname -m)
+# TODO: this should all move to the configure script
+# TODO: Makefile.config: configure; ./$<
+# TODO: include Makefile.config
```

Comment:
You can check out the [configure](https://github.com/project-everest/hacl-star/blob/franziskus/dist-build-update/dist/evercrypt/configure) and [Makefile](https://github.com/project-everest/hacl-star/blob/franziskus/dist-build-update/dist/evercrypt/Makefile) that do these things already. They are pretty aggressive in consolidating things. I'll leave it up to you if you want to incorporate these changes in this PR or if we do something like this in a follow up.

Reply:
I'd like to do a follow-up if that's ok... since it's likely going to require a few tweaks.

I looked at your changes and they seem good. From a quick glance at your files, I only have two remarks:
- CFLAGS_{128,256} should be determine by the configure script, shouldn't they?
- I'd like to disable -march=native,-mtune=native by default -- this is causing quite a few problems on windows where people tend to share binaries, and if compiled on a host that has a processor that's too recent, it'll generate illegal instructions errors -- note that openssl doesn't enable -march=native,-mtune=native by default

Do you mind opening up a PR to follow up?

Reply:
(If I open the PR I can't review it)

Reply:
Yeah let's do this in a follow up. This was more meant as a note to follow up on 🙂 
I'd like to decide on the route first, then I'll update the branch and do a PR.